docs/shoot_overlay_network.md

```diff
@@ -0,0 +1,84 @@
+# Enable / disable overlay network for shoots with Calico
+Gardener can be used with or without the overlay network. 
+
+Starting versions:
+- [provider-gcp@v1.25.0](https://github.com/gardener/gardener-extension-provider-gcp/releases/tag/v1.25.0)
+- [provider-alicloud@v1.40.0](https://github.com/gardener/gardener-extension-provider-alicloud/commit/3f3942635c850b443309697c51f66a605fbddc4b)
+- [provider-aws@v1.38.2](https://github.com/gardener/gardener-extension-provider-aws/releases/tag/v1.38.2)
+
+The default configuration of shoot clusters is without overlay network.
+
+## Understanding overlay network
+The Overlay networking permits the routing of packets between multiples pods located on multiple nodes, even is the pod and the node network are not the same.
+
+This is done through the encapsulation of pod packets in the node network so that the routing can be done as usual. We use `ipip` encapsulation with calico in case the overlay network is enabled. This (simply put) sends an IP packet as workload in another IP packet.
+
+![](./assets/Overlay-Network.drawio.png)
+
+In order to simplify the troubleshooting of problems and reducing the latency of packets travelling between nodes, the no overlay network is enabled by default for extension default as stated above for all new clusters.
+
+![](./assets/No-Overlay-Network.drawio.png)
+
+This means that the routing is done directly through the VPC routing table. Basically, when a new node it created, it is assigned a slice (usually a /24) of the pod network. All future pods in that node are going to be in this slice. Then, the cloud-controller-manager updated the cloud provider router to add the new route (all packets with in the network slice as destination should go to that node).
+
+This has the advantage of:
+- Doing less work for the node as encapsulation takes some CPU cycles.
+- The maximum transmission unit (MTU) is slightly bigger resulting in slightly better performance, i.e. potentially more workload bytes per packet.
+- More direct and simpler setup, which makes the problems much easier to troubleshoot.
+
+## Enabling the overlay network
+In certain cases, the overlay network might be preferable if, for example, the customer wants to create multiple cluster in the same VPC. 
+
+**In that case (multiple shoot in the same VPC), if the pod's network is not configured properly, there is a very strong chance that some pod IP address might overlay, which is going to cause all sorts of funny problems.** So, if someone asks you how to avoid that, they need to make sure that the podCIDR for each shoot **do not overlap with each other**.
```

Comment:
```suggestion
**In that case (multiple shoot in the same VPC), if the pod's network is not configured properly, there is a very strong chance that some pod IP address might overlap, which is going to cause all sorts of funny problems.** So, if someone asks you how to avoid that, they need to make sure that the podCIDR for each shoot **do not overlap with each other**.
```